internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -739,6 +739,41 @@ func NewCoreContainer(cr *operatorv1beta1.Cryostat, specs *ServiceSpecs, imageTa
 		}
 	}
 
+	disableBuiltInDiscovery := cr.Spec.TargetDiscoveryOptions != nil && cr.Spec.TargetDiscoveryOptions.BuiltInDiscoveryDisabled
+	if disableBuiltInDiscovery {
+		envs = append(envs, corev1.EnvVar{
+			Name:  "CRYOSTAT_DISABLE_BUILTIN_DISCOVERY",
+			Value: fmt.Sprintf("%v", disableBuiltInDiscovery),
```

Comment:
As mentioned elsewhere, if this env var is set with any non-empty value then it causes the built-ins to be disabled.

I would suggest actually hardcoding a value like `"true"` since this is wrapped in `if disableBuiltInDiscovery`. The way it's written now looks like an easy refactoring for someone to come "clean up" later on by removing the `if`, which would actually break the behaviour and always disable built-ins.

Reply:
Right, I could set it as `strconv.FormatBool(true)`, then if the way this var is checked changed later, we could replace with variable `disableBuiltInDiscovery `.

Reply:
This currently reads `fmt.Sprintf("%v", strconv.FormatBool(true))`. The `Sprintf` is unnecessary in this case because `strconv.FormatBool(true)` returns the string `"true"`. I'm with Andrew on this though, since this is always true, there's no need to call a function, `Value: "true"` makes most sense.

Reply:
oh right! I will remove it and just use `true` literal.